base/collection.go

```diff
@@ -388,30 +388,30 @@ func (b *GocbV2Bucket) Flush() error {
 
 }
 
-// BucketItemCount first tries to retrieve an accurate bucket count via N1QL,
-// but falls back to the REST API if that cannot be done (when there's no index to count all items in a bucket)
-func (b *GocbV2Bucket) BucketItemCount() (itemCount int, err error) {
-	dataStoreNames, err := b.ListDataStores()
+// BucketItemCount returns the amount of items in a bucket as per rest API.
+func (b *GocbV2Bucket) BucketItemCount() (int, error) {
+	type CurrItemsStats struct {
+		CurrItems int64 `json:"curr_items"`
```

Comment:
`curr_items` on the below URL, at least for me, returns a slice of samples, not just a single number.

```
"curr_items":[4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4],
```

I think we may need to use this API:
`https://docs.couchbase.com/server/current/rest-api/rest-buckets-summary.html`
and look up `basicStats.itemCount` instead